lib/index.js

```diff
@@ -17,183 +17,174 @@ internals.defaults = {
     port: 6379
 };
 
+module.exports = class Connection {
+    constructor(options) {
 
-exports = module.exports = internals.Connection = function (options) {
+        assert(this.constructor === Connection, 'Redis cache client must be instantiated using new');
 
-    Hoek.assert(this.constructor === internals.Connection, 'Redis cache client must be instantiated using new');
-
-    this.settings = Object.assign({}, internals.defaults, options);
-    this.client = null;
-    return this;
-};
-
-
-// Async
-internals.Connection.prototype.start = function () {
-
-    if (this.settings.client) {
-        this.client = this.settings.client;
+        this.settings = Object.assign({}, internals.defaults, options);
+        this.client = null;
+        return this;
     }
 
-    if (this.client) {
-        return Promise.resolve();
-    }
+    async start() {
 
-    // Return a promise that is resolved when everything is ready
-    return new Promise((resolve, reject) => {
+        if (this.settings.client) {
+            this.client = this.settings.client;
+        }
 
-        let client;
+        if (this.client) {
+            return;
+        }
 
-        const options = {
-            password: this.settings.password,
-            db: this.settings.database || this.settings.db,
-            tls: this.settings.tls
-        };
+        // Return a promise that is resolved when everything is ready
+        return await new Promise((resolve, reject) => {
 
-        if (this.settings.sentinels && this.settings.sentinels.length) {
-            options.sentinels = this.settings.sentinels;
-            options.name = this.settings.sentinelName;
-            client = Redis.createClient(options);
-        }
-        else if (this.settings.url) {
-            client = Redis.createClient(this.settings.url, options);
-        }
-        else if (this.settings.socket) {
-            client = Redis.createClient(this.settings.socket, options);
-        }
-        else {
-            client = Redis.createClient(this.settings.port, this.settings.host, options);
-        }
+            let client;
 
-        // Listen to errors
-        client.on('error', (err) => {
+            const options = {
+                password: this.settings.password,
+                db: this.settings.database || this.settings.db,
+                tls: this.settings.tls
+            };
 
-            if (!this.client) {             // Failed to connect
-                client.end(false);
-                return reject(err);
+            if (this.settings.sentinels && this.settings.sentinels.length) {
+                options.sentinels = this.settings.sentinels;
+                options.name = this.settings.sentinelName;
+                client = Redis.createClient(options);
```

Comment:
Could we make use of `new Redis()` instead of the deprecated `Redis.createClient()`?